---
layout: post
title: GSS 2019 Preparations - Differential Expression between infected and uninfected
date: '2019-11-12'
category: bairdi
tags: [GSS2019, DESeq2]
---
Today, I continued to prepare a presentation for GSS 2019 (this Thursday - my talk is at 1:30pm). I met with Steven and discussed a plan for results, and he shared with me the analyses he's done during the past couple of days. Details of results and what I've done in the post. I also tried doing edgeR stuff yesterday and didn't get super far, but details will be at the end of the post for my future self (won't do edgeR for GSS). 

### Stuff SR did (as far as I can understand at the moment)
#### [sr320/nb-2019/C_bairdi/11-Deseq2.html](http://htmlpreview.github.io/?https://github.com/sr320/nb-2019/blob/master/C_bairdi/11-Deseq2.html)

He joined three files      
1. [Abundance-merge.txt](https://raw.githubusercontent.com/sr320/nb-2019/master/C_bairdi/analyses/Abundance-merge.txt) - made from the outputs I made in `kallisto` ([bairdisamples-kallisto-updates](https://grace-ac.github.io/bairdisamples-kallisto-updates/)).
2. [sr320/nb-2019/Crab_DEGlist.txt](https://raw.githubusercontent.com/sr320/nb-2019/master/C_bairdi/analyses/Crab_DEGlist.txt) - a list of all the Differentially Expressed Genes 
3. [Cb_v1_blastx_sp_imac.tab](https://raw.githubusercontent.com/sr320/nb-2019/master/C_bairdi/analyses/Cb_v1_blastx_sp_imac.tab) - which is the blastoutput from the first assembled transcriptome against uniprot

The combined files resulted in: [bigtable.txt](https://raw.githubusercontent.com/sr320/nb-2019/master/C_bairdi/analyses/bigtable.txt)

#### [sr320/nb-2019/C_bairdi/11-Deseq2.html](http://htmlpreview.github.io/?https://github.com/sr320/nb-2019/blob/master/C_bairdi/11-Deseq2.html)

In the above link, Steven used DESeq2 to get the differentially expressed genes from the [Abundance-merge.txt](https://raw.githubusercontent.com/sr320/nb-2019/master/C_bairdi/analyses/Abundance-merge.txt) table. 

DEGs (p-value < 0.1):     
```
## out of 137634 with nonzero total read count
## adjusted p-value < 0.1
## LFC > 0 (up)       : 4529, 3.3%
## LFC < 0 (down)     : 59567, 43%
## outliers [1]       : 858, 0.62%
## low counts [2]     : 10635, 7.7%
## (mean count < 1)
## [1] see 'cooksCutoff' argument of ?results
## [2] see 'independentFiltering' argument of ?results
```

- Total DEGs with adj p-value <0.1 = 64096       
- DEGs in the infected group (down; negative LFC values) = 59567
- DEGs in the uninfected group (up; positive LFC values) = 4529

DEGs (p-value < 0.05):      
```
## out of 137634 with nonzero total read count
## adjusted p-value < 0.05
## LFC > 0 (up)       : 3062, 2.2%
## LFC < 0 (down)     : 55406, 40%
## outliers [1]       : 858, 0.62%
## low counts [2]     : 13314, 9.7%
## (mean count < 1)
## [1] see 'cooksCutoff' argument of ?results
## [2] see 'independentFiltering' argument of ?results
```

- Total DEGs with adj p-value <0.05 = 58468
- DEGs in the infected group (down; negative LFC values) = 55406
- DEGs in the uninfected group (up; positive LFC values) = 3062


